### DISCUSSION TITLE
'Expect' assertion to check the element's absence doesn't work

### DISCUSSION THREAD

**Discussion by sureshautomation** [cypress-io/cypress#14648] [Questions and Help] [answered]
hi, i have created custom commands for existence assertion using .expect() 

Below is the command sample:
**Cypress.Commands.add('isExist', (element, elementName, exist = true, wait = 10000) => {
  if (exist) {
    cy.xpath(element, {timeout: wait}).then((e) => {
      expect(e, elementName).to.exist;
    });
  } else {
    cy.xpath(element, {timeout: wait}).then((e) => {
      expect(e, elementName).not.to.exist;
    });
  }
});**

then i use it like below
cy.isExist('locator', 'locatorName', true, 12000);

now, it works correctly when 'exist' is true but it fails when 'exist' is false.
for ex: cy.isExist('locator', 'locatorName', false); ---> this line throws below error

**AssertionError
Timed out retrying: Expected to find element: undefined, but never found it.**

my concern is, this element is not gonna be exist in the DOM and i am already asserting that the element is not exist, then why is it failing with the reason as "Timed out retrying"?

**Comment by bahmutov** [comment, marked as answer]
The `cy.xpath` already checks if the element exists, thus by the time you get the element it is too late. You might want to write your own xpath command this such option following the blog post https://glebbahmutov.com/blog/writing-custom-cypress-command/

**Reply by sureshautomation** [reply]
thanks for your response. but am not facing this issue when i use cy.xpath(locator).should('not.exist'). i get the error only when i assert with 'expect'

**Reply by sureshautomation** [reply]
just an update.
i have updated **then(e)** with **should(e)**, then it is working as expected and no timeout error shown.
**cy.xpath(element, {timeout: wait}).should((e) => {
expect(e, elementName).not.to.exist;
});**

**Comment by bahmutov** [comment]
Perfect
